code/weapon/weapons.cpp

```diff
@@ -2382,10 +2382,15 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		ci.dinky = false;
 		required_string("+Armor Type:");
 			stuff_string(fname, F_NAME, NAME_LENGTH);
-		armor_index = armor_type_get_idx(fname);
-		if (armor_index < 0) {
-			Warning(LOCATION, "Armor type '%s' not found for conditional impact in weapon %s!", fname, wip->name);
-		}
+		if (*fname == *"NO ARMOR") {
```

Comment:
C-style strings require a comparison function, such as `stricmp`, which stands for case-insensitive string compare.  To test for equality, check `stricmp(string1, string2) == 0`, or more succinctly, `!stricmp(string1, string2)`.

More information here.  This covers `strcmp` which is the case-sensitive version, but `stricmp` works the same way:
https://en.cppreference.com/w/cpp/string/byte/strcmp